src/libcore/num/mod.rs

```diff
@@ -464,6 +452,30 @@ macro_rules! uint_impl {
                     v => Some(self / v),
                 }
             }
+
+            #[inline]
+            fn pow(self, mut exp: u32) -> Self {
+                let mut base = self;
+                let mut acc: Self = Int::one();
+                let mut did_base_overflow = false;
+
+                while exp > 0 {
+                    if (exp & 1) == 1 {
+                        acc = acc * base;
+                        if did_base_overflow {
+                            panic!("overflow detected");
+                        }
```

Comment:
I'm a little confused by these overflow semantics. How come the overflow is only registered on the next iteration of the loop? Additionally, how come overflow is registered at all? In theory this has the same overflow semantics as `a * b` which is actually listed below (with `base * base`) so isn't overflow naturally detected?